### ISSUE TITLE
First four characters are deleted from folder name when "Keep folder structure" is selected

### ISSUE THREAD

**Issue by deltz025** [Lymphatus/caesium-image-compressor#84]
This bug may be Mac-specific, as I'm not at my Windows machine to try replicating it. I'll see about it later and add an update. 

When compressing multiple folders, the first four characters of each folder's name is omitted. Folders with less than five characters in their title are deleted altogether, with pictures inside them being sent to the selected output folder. 

![Screen Shot 2021-11-18 at 2 28 06 PM](https://user-images.githubusercontent.com/49649081/142485098-a17342f1-afcf-407a-80b3-3d2a77bc6ad6.png)
![Screen Shot 2021-11-18 at 2 33 58 PM](https://user-images.githubusercontent.com/49649081/142485099-56214ed8-36da-4c30-8a69-b2e0a0f47f70.png)

I attempted to recreate this bug by creating a folder with numerous subfolders of varying name lengths, then running Caesium on the entire folder, but something different happened. This time, it recreated the path to that folder within the folder. This happened when I dragged the parent folder into Caesium, as well as when I dragged all the subfolders in. 
![Screen_Shot_2021-11-18_at_2 46 44_PM](https://user-images.githubusercontent.com/49649081/142486805-66060fee-1f4c-495c-b0ce-990f8a968eaf.png)

**Comment by Lymphatus** [comment]
That's odd. I develop mostly on Mac and this didn't really happen to me. I'll try to reproduce the issue though.

About the second attempt: in that case Caesium tried to recreate the structure of the multiple folders, but failed for some reason (it should not recreate the whole path starting from /).

I'll investigate, thanks.

**Comment by Lymphatus** [comment]
Are the original images on a different drive than the destination? I have some weird behaviour in this case.

**Comment by deltz025** [comment]
They are. I was compressing a series of folders from a company server when I experienced the bug. With that in mind, I was able to recreate the bug by trying to compress the same folders, but it doesn't happen consistently. 

Running just a handful of the folders from the server resulted in the same bug, but then running that exact same set again didn't. Running the entire directory of folders, which caused the initial bug, also failed to recreate the bug.
